backend/endpoints/news/route.py

```diff
@@ -0,0 +1,34 @@
+from dataclasses import dataclass
+from ninja import Router
+from django.http.request import HttpRequest
+
+from lib.polygon_api import PolygonAPI, normalise_scores
+
+router = Router()
+
+@dataclass
+class TickerArticleDataclass:
+    title: str
+    description: str
+    url: str
+    date: str
+    publisher: str
+    score: float
+    tickers: list[str]
+
+@dataclass
+class NewsResponse:
+    news: list[TickerArticleDataclass]
+
+@router.get("/news/", response=NewsResponse)
+def news(request: HttpRequest, ticker: str, n=20) -> NewsResponse:
+    """
+    Gets the list of the n most recent articles about ticker
+    """
+    api = PolygonAPI()
+    news = normalise_scores(api.get_news(ticker, n))
```

Comment:
Instead of getting news from API every time, we should have a background thread that periodically requests latest news for all tickers, and updates the database if it doesn't exist. That way we can properly model relationships between publishers, news articles, trades etc.

Reply:
Also need to return a sorted list weighted by recency and objectivity. Probably worth also adding objectivity and polarity scores to articles.